rcl/test/rcl/test_get_actual_qos.cpp

```diff
@@ -151,7 +158,61 @@ TEST_P_RMW(TestGetActualQoS, test_publisher_get_qos_settings) {
   rcl_reset_error();
 }
 
-static constexpr rmw_qos_profile_t non_default_qos_profile()
+
+class TEST_FIXTURE_P_RMW(TestSubscriptionGetActualQoS)
+  : public TEST_FIXTURE_P_RMW(TestGetActualQoS) {};
+
+TEST_P_RMW(TestSubscriptionGetActualQoS, test_subscription_get_qos_settings) {
+  TestParameters parameters = GetParam();
+  std::string topic_name("/test_subscription_get_qos_settings");
+  rcl_ret_t ret;
+
+  rcl_subscription_t pub = rcl_get_zero_initialized_subscription();
+  rcl_subscription_options_t pub_ops = rcl_subscription_get_default_options();
+  pub_ops.qos = parameters.qos_to_set;
+  auto ts = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BasicTypes);
+  ret = rcl_subscription_init(&pub, this->node_ptr, ts, topic_name.c_str(), &pub_ops);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  const rmw_qos_profile_t * qos;
+  qos = rcl_subscription_get_actual_qos(&pub);
+  EXPECT_NE(nullptr, qos) << rcl_get_error_string().str;
+  EXPECT_EQ(
+    qos->history,
+    parameters.qos_expected.history);
+  EXPECT_EQ(
+    qos->depth,
+    parameters.qos_expected.depth);
+  EXPECT_EQ(
+    qos->reliability,
+    parameters.qos_expected.reliability);
+  EXPECT_EQ(
+    qos->durability,
+    parameters.qos_expected.durability);
+  EXPECT_EQ(
+    qos->liveliness,
+    parameters.qos_expected.liveliness);
+  EXPECT_EQ(
+    qos->avoid_ros_namespace_conventions,
+    parameters.qos_expected.avoid_ros_namespace_conventions);
+
```

Comment:
It would be good to add similar `EXPECT_EQ` statements for `liveliness_lease_duration`, `deadline` and `lifespan`.

Reply:
Every `rmw` implementation's default values for `deadline`, `lifespan`, and `liveliness_lease_duration` seems to be different. I'll use an inequality test instead.

Reply:
Done.